ingestion/src/metadata/profiler/source/database/base/profiler_source.py

```diff
@@ -122,27 +121,95 @@ def _copy_service_config(
 
         return config_copy
 
+    # TODO: Ask... Do we need the db_service/profiler_config here for any reason?
     def create_profiler_interface(
         self,
         entity: Table,
         config: Optional[TableConfig],
         profiler_config: Optional[ProfilerProcessorConfig],
-        schema_entity: Optional[DatabaseSchema],
-        database_entity: Optional[Database],
+        schema_entity: DatabaseSchema,
+        database_entity: Database,
         db_service: Optional[DatabaseService],
     ) -> ProfilerInterface:
-        """Create sqlalchemy profiler interface"""
+        """Create the appropriate profiler interface based on processing engine."""
+
+        # TODO: Any reason to instantiate this here instead of __init__? Not sure I'm getting the difference
         self.source_config = DatabaseServiceProfilerPipeline.model_validate(
             self.config.source.sourceConfig.config
         )
```

Comment:
We can try to move it to the __init__ and see if we break anything. Unusre why it is instantiated here.